src/main/java/com/intuit/graphql/orchestrator/schema/transform/KeyTransformerPostMerge.java

```diff
@@ -0,0 +1,40 @@
+package com.intuit.graphql.orchestrator.schema.transform;
+
+import static com.intuit.graphql.orchestrator.utils.XtextGraphUtils.addToCodeRegistry;
+
+import com.intuit.graphql.graphQL.FieldDefinition;
+import com.intuit.graphql.orchestrator.apollofederation.EntityExtensionContext;
+import com.intuit.graphql.orchestrator.xtext.DataFetcherContext;
+import com.intuit.graphql.orchestrator.xtext.FieldContext;
+import com.intuit.graphql.orchestrator.xtext.XtextGraph;
+
+public class KeyTransformerPostMerge implements Transformer<XtextGraph, XtextGraph> {
+
+  @Override
+  public XtextGraph transform(XtextGraph source) {
+    source.getEntityFieldExtensionsMap().keySet().stream()
```

Comment:
would it make more sense for this to be a foreach loop on map itself? I see you are using the stream to just filter and the filter is also couple of lines. Also, it will avoid doing this twice
`EntityExtensionContext entityExtensionContext = source.getEntityFieldExtensionsMap().get(fieldCoordinates);`

Reply:
we changed the data structure.  latest code is [here](https://github.com/intuit/graphql-orchestrator-java/pull/47/files#diff-ed6c24b0102148dd5dfec8523a39bfb27fbccc8b25365f0a9cfbf5c94bf2cf99R29)